pkg/cloud/cluster.go

```diff
@@ -92,6 +86,44 @@ func (c *client) GetOrCreateCluster(csCluster *infrav1.CloudStackCluster) (retEr
 	return nil
 }
 
+func (c *client) ResolveDomainAndAccount(csCluster *infrav1.CloudStackCluster) error {
+	if (csCluster.Spec.Domain != "" && csCluster.Spec.Account == "") ||
+		(csCluster.Spec.Domain == "" && csCluster.Spec.Account != "") {
+		return errors.Errorf("Both domain and account must be specified or none of them must be specified")
+	}
+
+	if csCluster.Spec.Domain != "" && csCluster.Spec.Account != "" {
+		p := c.cs.Domain.NewListDomainsParams()
+		p.SetListall(true)
```

Comment:
Lets at least narrow it by name since it's so easy to do. p.SetName(Domain) or whatever the method is.

Reply:
Otherwise, looks good overall.

Reply:
If domain: value is a fqn then setName() make make the list domains return nothing.